### DISCUSSION TITLE
Use zstd compression for backups?

### DISCUSSION THREAD

**Discussion by LordHelmchen666** [AzuraCast/AzuraCast#4870] [Ideas]
I have seen Plesk is now using ZStandard to compress backups. So i asked myself what are the benefits and found the answer. Compared to gzip and zip it is faster in compressing and decompressing, uses less CPU and Memory.
Tried it on my system with result. zip takes around 40 min, filesize 40,8GB. gzip 38min, 40.7GB. zstd  15min, 38,9GB.
So maybe it is worth to use it also for backups in Azuracast.

You can use it very simple with tar
`tar -I "zstd -T0" -cf backup-file.tzst folder-to-backu`

**Comment by BusterNeece** [comment]
That's a significant performance improvement...will certainly investigate this.

**Comment by BusterNeece** [comment]
@LordHelmchen666 I'm seeing a big performance bump in my own tests, so I've added the option for zstd-backed backups and restores into both Docker and Ansible installations; if you create a backup with the extension `.tzst`, it'll use that library to generate the file, and my own experience shows it being up to 10x faster :o

**Reply by LordHelmchen666** [reply]
That is great. Can you please add "v" to tar command (-cvf) so we get little verbose.

**Reply by BusterNeece** [reply]
Done!

**Comment by LordHelmchen666** [comment]
@SlvrEagle23 Could you please make chose able for automatic backups? Maybe in that style
```
which compression to use for automatic backup?
 [ ] zip
 [ ] tgz
 [ ] zstd
```
Because size is much smaller in zstd compared to zip.
manuell.tzst  | 51.65 MB 
automatic_backup_20220427_023206.zip | 112.74 MB
